bee-network/bee-autopeering/src/config.rs

```diff
@@ -0,0 +1,329 @@
+    #[serde(rename = "dropNeighborsOnSaltUpdate")]
+    pub drop_neighbors_on_salt_update: Option<bool>,
+}
+
+impl AutopeeringConfigJsonBuilder {
+    /// Builds the actual `AutopeeringConfig`.
+    pub fn finish(self) -> AutopeeringConfig {
+        AutopeeringConfig {
+            enabled: self.enabled,
+            bind_addr: self.bind_addr,
+            entry_nodes: self.entry_nodes,
+            entry_nodes_prefer_ipv6: self.entry_nodes_prefer_ipv6.unwrap_or(ENTRYNODES_PREFER_IPV6_DEFAULT),
+            run_as_entry_node: self.run_as_entry_node.unwrap_or(RUN_AS_ENTRYNODE_DEFAULT),
+            drop_neighbors_on_salt_update: self
+                .drop_neighbors_on_salt_update
+                .unwrap_or(DROP_NEIGHBORS_ON_SALT_UPDATE_DEFAULT),
+        }
+    }
+}
+
+impl Default for AutopeeringConfigJsonBuilder {
+    fn default() -> Self {
+        Self {
+            enabled: false,
+            bind_addr: SocketAddr::new(AUTOPEERING_BIND_ADDR_DEFAULT, AUTOPEERING_BIND_PORT_DEFAULT),
+            entry_nodes: Vec::default(),
+            entry_nodes_prefer_ipv6: Some(false),
+            run_as_entry_node: Some(false),
+            drop_neighbors_on_salt_update: Some(false),
+        }
+    }
+}
+
+/// The autopeering config TOML builder.
+///
+/// Note: Fields will be snake-case formatted.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(test, derive(Eq, PartialEq))]
+#[serde(rename = "autopeering")]
+pub struct AutopeeringConfigTomlBuilder {
+    /// Wether autopeering should be enabled.
+    pub enabled: bool,
+    /// The bind address for the server.
+    #[serde(rename = "bind_address")]
+    pub bind_addr: SocketAddr,
+    /// The entry nodes for bootstrapping.
+    pub entry_nodes: Vec<AutopeeringMultiaddr>,
+    /// Whether `Ipv4` or `Ipv6` should be preferred in case a hostname supports both.
+    pub entry_nodes_prefer_ipv6: Option<bool>,
+    /// Whether the node should run as an entry node.
+    pub run_as_entry_node: Option<bool>,
+    /// Whether all neighbors should be disconnected from when the salts are updated.
+    pub drop_neighbors_on_salt_update: Option<bool>,
+}
+
+impl AutopeeringConfigTomlBuilder {
+    /// Builds the actual `AutopeeringConfig`.
+    pub fn finish(self) -> AutopeeringConfig {
+        AutopeeringConfig {
+            enabled: self.enabled,
+            bind_addr: self.bind_addr,
+            entry_nodes: self.entry_nodes,
+            entry_nodes_prefer_ipv6: self.entry_nodes_prefer_ipv6.unwrap_or(ENTRYNODES_PREFER_IPV6_DEFAULT),
+            run_as_entry_node: self.run_as_entry_node.unwrap_or(RUN_AS_ENTRYNODE_DEFAULT),
+            drop_neighbors_on_salt_update: self
+                .drop_neighbors_on_salt_update
+                .unwrap_or(DROP_NEIGHBORS_ON_SALT_UPDATE_DEFAULT),
+        }
+    }
+}
+
+impl Default for AutopeeringConfigTomlBuilder {
+    fn default() -> Self {
+        let json_builder = AutopeeringConfigJsonBuilder::default();
```

Comment:
Not sure I like this tbh, it's actually more verbose and a little bit weird

Reply:
yea, it's weird. I had the same feeling when writing this...but in the end I am kinda pragmatic, too :trollface:

Reply:
Changed.